source/simulator/core.cc

```diff
@@ -75,6 +76,17 @@ namespace aspect
 
       return false;
     }
+
+    template <int dim>
+    std::vector<dealii::Variable<dim> > get_variables(const Parameters<dim> &parameters,
+                                                      SimulatorSignals<dim> &signals)
```

Comment:
The function could also use a piece of documentation :-)

I think there is relatively little value in the function, though. It has four lines, and it requires two lines to call it. If you change the signal so that it takes a (const) list of variables and returns a (possibly modified) list of variables, then you can just chain the calls to `construct_variables` and `edit_finite_element_variables` and inline it all into the place where you currently call it.